driver/src/main/java/org/neo4j/driver/internal/async/NetworkConnection.java

```diff
@@ -208,73 +217,49 @@ private void writeResetMessageIfNeeded(ResponseHandler resetHandler, boolean isS
         });
     }
 
-    private void flushInEventLoop() {
-        channel.eventLoop().execute(() -> {
-            channel.flush();
-            registerConnectionReadTimeout(channel);
-        });
-    }
-
     private void writeMessageInEventLoop(Message message, ResponseHandler handler, boolean flush) {
```

Comment:
I can't find the unit test code that tests this changes. It might worth to add some unit to `write` and `writeAndFlush` methods to cover cases where there is a `TerminatorAwareExecutor` and it's terminated or not. Also, when the message are in the list of out of the desired list, and so on.

Reply:
Added more tests.